### PR TITLE
[REVIEW] enable per-thread default stream in gpu ci build

### PR DESCRIPTION
**Pull request by rongou**
Always build and test with per-thread default stream enabled, as discussed in https://github.com/rapidsai/cudf/issues/5614.

@harrism @jrhemstad @kkraus14

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by rongou** [comment]
rerun tests

File: build.sh [hunk 1]
```diff
@@ -18,15 +18,16 @@ ARGS=$*
...
+VALIDARGS="clean librmm rmm -v -g -n -s -p -h"
+HELP="$0 [clean] [librmm] [rmm] [-v] [-g] [-n] [-s] [-p] [-h]
    clean  - remove all existing build artifacts and configuration (start over)
    librmm - build and install the librmm C++ code
    rmm    - build and install the rmm Python package
    -v     - verbose build mode
    -g     - build for debug
    -n     - no install step
    -s     - statically link against cudart
+   -p     - enable per-thread default stream
```
Comment (kkraus14) [inline build.sh:30]: Can we make this consistent between rmm and cudf? I would prefer `--ptds` personally.

Comment (rongou) [reply build.sh]: Done.

**Pull request merged**